app/models/linkset.rb

```diff
@@ -12,6 +12,14 @@ def empty?
   end
 
   def update_attributes_from_gem_specification!(spec)
-    self.update_attributes!(:home => spec.homepage)
+    meta = spec.metadata || {}
+    self.update_attributes!({
+      :home => spec.homepage,
+      :wiki => meta["wiki_uri"],
+      :docs => meta["documentation_uri"],
+      :mail => meta["mailing_list_uri"],
+      :code => meta["source_code_uri"],
+      :bugs => meta["bug_tracker_uri"]
+    }.delete_if { |_n, v| v.nil? })
```

Comment:
the `metadata` is freeform, but the problem is it is limited, you can only define strings and with limited size, anothe thing is those urls are already handled, so it's basically new way of specifying this urls, if you want more data then it would be part of new request